static/src/javascripts/projects/commercial/modules/article-body-adverts.js

```diff
@@ -0,0 +1,209 @@
+        },
+        filter(slot) {
+            if (
+                !prevSlot ||
+                Math.abs(slot.top - prevSlot.top) - adSizes.mpu.height >=
+                    this.selectors[' .ad-slot'].minBelow
+            ) {
+                prevSlot = slot;
+                return true;
+            }
+            return false;
+        },
+    };
+};
+
+const getAltRules = () => {
+    const altRules = getRules();
+    altRules.selectors = {
+        ' .ad-slot': {
+            minAbove: 500,
+            minBelow: 500,
+        },
+    };
+    return altRules;
+};
+
+const getLongArticleRules = () => {
+    const longArticleRules = isOffsetingAds ? getAltRules() : getRules();
+    const viewportHeight = detect.getViewport().height;
+    longArticleRules.selectors[' .ad-slot'].minAbove = viewportHeight;
+    longArticleRules.selectors[' .ad-slot'].minBelow = viewportHeight;
+    return longArticleRules;
+};
+
+const addInlineAds = () =>
+    addArticleAds(
+        2,
+        isOffsetingAds ? getAltRules() : getRules()
+    ).then(countAdded => {
+        if (countAdded === 2) {
+            return addArticleAds(8, getLongArticleRules()).then(
+                innerCountAdded => 2 + innerCountAdded
+            );
+        }
+        return countAdded;
+    });
+
+const getInlineMerchRules = () => {
+    const inlineMerchRules = getRules();
+    inlineMerchRules.minAbove = 300;
+    inlineMerchRules.selectors[' > h2'].minAbove = 100;
+    inlineMerchRules.selectors[
+        ' > :not(p):not(h2):not(.ad-slot)'
+    ].minAbove = 200;
+    return inlineMerchRules;
+};
+
+const addInlineMerchAd = () =>
+    spaceFiller.fillSpace(
+        getInlineMerchRules(),
+        paras => insertAdAtPara(paras[0], 'im', 'im').then(() => 1),
+        {
+            waitForImages: true,
+            waitForLinks: true,
+            waitForInteractives: true,
+        }
+    );
+
+const waitForMerch = (countAdded: number) =>
+    countAdded === 1 ? trackAdRender('dfp-ad--im') : Promise.resolve();
+
+const init = () => {
+    if (!commercialFeatures.articleBodyAdverts) {
+        return Promise.resolve(false);
+    }
+
+    bodyAds = 0;
+    replaceTopSlot = detect.isBreakpoint({
+        max: 'phablet',
+    });
+    getSlotName = replaceTopSlot ? getSlotNameForMobile : getSlotNameForDesktop;
+    getSlotType = replaceTopSlot ? getSlotTypeForMobile : getSlotTypeForDesktop;
+
+    if (config.page.hasInlineMerchandise) {
+        const im = addInlineMerchAd();
+        // Whether an inline merch has been inserted or not,
+        // we still want to try to insert inline MPUs. But
+        // we must wait for DFP to return, since if the merch
+        // component is empty, it might completely change the
+        // positions where we insert those MPUs.
+        im.then(waitForMerch).then(addInlineAds);
+        return im;
+    }
+
+    addInlineAds();
+    return Promise.resolve(true);
+};
+
+export default {
+    init,
+
+    '@@tests': {
```

Comment:
that's what the `_` export is in effect, except it will be tree-shook. happy to rename to `test` if everyone prefers it, but we should pick _one_ and stick with it :)